main.js

```diff
@@ -62,101 +64,116 @@ function createWindow() {
 	mainWindow = new BrowserWindow(browserWindowOptions)
 
   // and load the index.html of the app.
-	mainWindow.loadURL('file://' + __dirname + '/index.html')
+	mainWindow.loadURL("file://" + __dirname + "/index.html")
 
   // Open the DevTools if run with "npm start dev"
 	if (DevelopmentMode) {
 		mainWindow.webContents.openDevTools()
 	}
 
   // Emitted when the window is closed.
-	mainWindow.on('closed', function () {
+	mainWindow.on("closed", function () {
     // Dereference the window object, usually you would store windows
     // in an array if your app supports multi windows, this is the time
     // when you should delete the corresponding element.
 		mainWindow = null
 	})
 }
 
-// Initilize the keyword spotter
-if (config && config.speech && !firstRun) {
-	var kwsProcess = spawn('node', ['./sonus.js'], { detached: false })
-  // Handel messages from node
-	kwsProcess.stderr.on('data', function (data) {
+function startSonus()
+{
+  // Initilize the keyword spotter
+	var kwsProcess = spawn("node", ["./sonus.js"], { detached: false })
+    // Handel messages from node
+	kwsProcess.stderr.on("data", function (data) {
 		var message = data.toString()
 		console.error("ERROR", message.substring(4))
 	})
 
-	kwsProcess.stdout.on('data', function (data) {
+	kwsProcess.stdout.on("data", function (data) {
 		var message = data.toString()
-		if (message.startsWith('!h:')) {
-			mainWindow.webContents.send('hotword', true)
-		} else if (message.startsWith('!p:')) {
-			mainWindow.webContents.send('partial-results', message.substring(4))
-		} else if (message.startsWith('!f:')) {
-			mainWindow.webContents.send('final-results', message.substring(4))
+		if (message.startsWith("!h:")) {
+			mainWindow.webContents.send("hotword", true)
+		} else if (message.startsWith("!p:")) {
+			mainWindow.webContents.send("partial-results", message.substring(4))
+		} else if (message.startsWith("!f:")) {
+			mainWindow.webContents.send("final-results", message.substring(4))
 		} else {
 			console.error(message.substring(3))
 		}
 	})
+    
+    // if we receive a closed event from the keyword spotter
+	kwsProcess.on("close", function() {
```

Comment:
We shouldn't need to restart Sonus anymore, it should handle things on its own:
https://github.com/evancohen/sonus/releases/tag/1.0.1

Reply:
no.. this is another bug,.. suddenly the sonus process just disappears..  not the arecord falling out from under it..

an example.. kill the node sonus task...   how do we recover?? we don't